bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/ImportHBaseSnapshotJob.java

```diff
@@ -0,0 +1,290 @@
+
+    if (setConfFromArgs(conf, otherArgs) != 0) return -1;
+
+    LOG.info(
+        "Using Arguments: "
+            + "\n"
+            + "snapshotName = "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + "\n"
+            + "snapshotDir = "
+            + conf.get(HConstants.HBASE_DIR)
+            + "\n"
+            + "bigtableName = "
+            + conf.get(TableOutputFormat.OUTPUT_TABLE)
+            + "\n"
+            + "tempDir = "
+            + conf.get(SNAPSHOT_RESTOREDIR_KEY)
+            + "\n"
+            + CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY
+            + " = "
+            + conf.get(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY)
+            + "\n");
+
+    Job job = createSubmittableJob(conf);
+    int successStatus = (job.waitForCompletion(true) ? 0 : 1);
+
+    Counters counters = job.getCounters();
+    long numRows = counters.findCounter(ScanCounter.NUM_ROWS).getValue();
+    long numCells = counters.findCounter(ScanCounter.NUM_CELLS).getValue();
+
+    System.out.println(("############# Num of ROWS imported= " + numRows));
+    System.out.println(("############# Num of CELLS imported= " + numCells));
+
+    return successStatus;
+  }
+
+  protected static int setConfFromArgs(Configuration conf, String[] args) {
+    // basic pre-condition check
+    if (args.length < 4) {
+      usage("Wrong number of arguments: " + args.length);
+      return -1;
+    }
+    if (conf.get(BigtableOptionsFactory.PROJECT_ID_KEY) == null) {
+      usage("Must specify the property " + BigtableOptionsFactory.PROJECT_ID_KEY);
+      return -1;
+    }
+    if (conf.get(BigtableOptionsFactory.INSTANCE_ID_KEY) == null) {
+      usage("Must specify the property " + BigtableOptionsFactory.INSTANCE_ID_KEY);
+      return -1;
+    }
+
+    // set arguments in configuration for ease of use
+    conf.set(SNAPSHOTNAME_KEY, args[0]);
+    conf.set(HConstants.HBASE_DIR, args[1]);
+    conf.set(TABLENAME_KEY, args[2]);
+    conf.set(SNAPSHOT_RESTOREDIR_KEY, args[3]);
+
+    // set default fs
+    conf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, conf.get(HConstants.HBASE_DIR));
+
+    // default job name
+    conf.setIfUnset(
+        IMPORT_SNAPSHOT_JOBNAME_KEY,
+        ImportHBaseSnapshotJob.class.getName()
+            + " - from snapshot "
+            + conf.get(SNAPSHOTNAME_KEY)
+            + " to "
+            + conf.get(TABLENAME_KEY));
+
+    // implicit bigtable configs
+    BigtableConfiguration.configure(
+        conf,
+        conf.get(BigtableOptionsFactory.PROJECT_ID_KEY),
+        conf.get(BigtableOptionsFactory.INSTANCE_ID_KEY),
+        conf.get(BigtableOptionsFactory.APP_PROFILE_ID_KEY, ""));
+    conf.setBooleanIfUnset(BigtableOptionsFactory.BIGTABLE_USE_TIMEOUTS_KEY, true);
+    conf.setIfUnset(
+        BigtableOptionsFactory.BIGTABLE_MUTATE_RPC_TIMEOUT_MS_KEY,
+        String.valueOf(BIGTABLE_MUTATE_RPC_TIMEOUT_MS_DEFAULT));
```

Comment:
why is this necessary?

Reply:
instead of relying on a default request timeout, the approach here is to increase this for batch mutations explicitly to indicate to users how to adjust this threshold. it may not be evident to end users that TableOutputFormat uses a bufferedmutator with this specific timeout config. what are the specs for the default request timeout if we remove this?

Reply:
it should follow java-bigtable's configs. Which are defined here:
https://github.com/googleapis/googleapis/blob/master/google/bigtable/v2/bigtable_grpc_service_config.json#L44

and augmented here:
https://github.com/googleapis/java-bigtable/blob/77e6f9ef11ad7cfa474eed1826fe5ad1e32f16a2/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/EnhancedBigtableStubSettings.java#L571-L598

If the intention is to give users an example how to change this, I think it should be moved to docs?

Reply:
done